java/src/org/openqa/selenium/json/Json.java

```diff
@@ -27,20 +27,117 @@
 import java.util.Map;
 
+/**
+ * The <b>Json</b> class is the entrypoint for the JSON processing features of the Selenium API.
+ * <p>
+ * These features include:
+ * <ul>
+ *   <li>Built-in serialization of primitives and collections.</li>
+ *   <li>Facilities to serialize custom data types:
+ *     <ul>
+ *       <li>Classes that declare getter methods that adhere to the
+ *         <a href="https://docs.oracle.com/javase/tutorial/javabeans/writing/index.html">JavaBean</a>
+ *         specification.</li>
+ *       <li>Classes that declare a {@code toJson()} method that returns primitives and collections.</li>
+ *     </ul>
+ *   </li>
+ *   <li>Build-in deserialization of primitives and collections.</li>
+ *   <li>Facilities to deserialize custom data types:
+ *     <ul>
+ *       <li>Classes that declare setter methods that adhere to the {@code JavaBean} specification.</li>
+ *       <li>Classes that declare a {@code fromJson(JsonInput)} method.</li>
+ *     </ul>
+ *   </li>
+ * </ul>
+ *
+ * The standard types supported by built-in processing:
+ * <ul>
+ *   <li><b>Numeric Types</b>:
+ *     <ul>
+ *       <li>{@link java.lang.Byte Byte}</li>
+ *       <li>{@link java.lang.Double Double}</li>
+ *       <li>{@link java.lang.Float Float}</li>
+ *       <li>{@link java.lang.Integer Integer}</li>
+ *       <li>{@link java.lang.Long Long}</li>
+ *       <li>{@link java.lang.Short Short}</li>
+ *     </ul>
+ *   </li>
+ *   <li><b>Collection Types</b>:
+ *     <ul>
+ *       <li>{@link java.util.List List}</li>
+ *       <li>{@link java.util.Set Set}</li>
+ *     </ul>
+ *   </li>
+ *   <li><b>Standard Java Types</b>:
+ *     <ul>
+ *       <li>{@link java.util.Map Map}</li>
+ *       <li>{@link java.lang.Boolean Boolean}</li>
+ *       <li>{@link java.lang.String String}</li>
+ *       <li>{@link java.lang.Enum Enum}</li>
+ *       <li>{@link java.net.URI URI}</li>
+ *       <li>{@link java.net.URL URL}</li>
+ *       <li>{@link java.util.UUID UUID}</li>
+ *       <li>{@link java.time.Instant Instant}</li>
+ *       <li>{@link java.lang.Object Object}</li>
+ *     </ul>
+ *   </li>
+ * </ul>
+ *
+ * The <b>Json</b> API also handles Selenium {@link org.openqa.selenium.Capabilities Capabilities} objects:
+ * <pre>
+ *   String json = "{\"browserName\":\"htmlunit\",\"version\":\"chrome\"}";
+ *   Capabilities capabilities = new Json().toType(json, Capabilities.class);
+ * </pre>
+ *<p>
+ * You can serialize objects for which no explicit coercer has been specified, and the <b>Json</b> API will use a
+ * generic process to provide best-effort JSON output. For the most predictable results, though, it's best to
+ * provide a {@code toJson()} method for the <b>Json</b> API to use for serialization. This is especially beneficial
+ * for objects that contain transient properties that should be omitted from the JSON output.
+ * <p>
+ * You can deserialize objects for which no explicit coercer has been defined via the <b>ObjectCoercer</b>. Note
```

Comment:
The entry points for users of the JSON package are all in the `Json` class, though people might need to interact with a `JsonInput`. We shouldn't be encouraging people to look at the coercers.

The right thing to do is to use `Json.toType()`, which because of its funky signature will attempt to return the right type of object. You can use a `TypeToken` to coerce to novel types. eg `Set<String> strings = Json.toType(input, new TypeToken<Set<String>>(){}.getType())`